esp/services/ws_fs/ws_fsService.cpp

```diff
@@ -1814,15 +1814,47 @@ bool CFileSprayEx::onGetDFUExceptions(IEspContext &context, IEspGetDFUExceptions
     return true;
 }
 
-void CFileSprayEx::readAndCheckSpraySourceReq(MemoryBuffer& srcxml, const char* srcIP, const char* srcPath,
+void CFileSprayEx::readAndCheckSpraySourceReq(MemoryBuffer& srcxml, const char* srcIP, const char* srcPath, const char* srcPlane,
     StringBuffer& sourceIPReq, StringBuffer& sourcePathReq)
 {
-    StringBuffer sourcePath(srcPath);
-    sourceIPReq.set(srcIP);
-    sourceIPReq.trim();
-    sourcePath.trim();
+    StringBuffer sourcePath;
+
     if(srcxml.length() == 0)
     {
+        if (!isEmptyString(srcPlane))
+        {
+            Owned<IPropertyTree> dropZone =  getDropZonePlane(srcPlane);
+            if (!dropZone)
+                throw MakeStringException(ECLWATCH_INVALID_INPUT, "Unknown landing zone: %s", srcPlane);
+            const char * dropZonePlanePath = dropZone->queryProp("@prefix");
+            if (isAbsolutePath(srcPath))
+            {
+                if (!startsWith(srcPath,dropZonePlanePath))
+                    throw MakeStringException(ECLWATCH_INVALID_INPUT, "Invalid source path");
+                sourcePath.append(srcPath).trim();
+            }
+            else
+            {
+                sourcePath.append(dropZonePlanePath);
+                addNonEmptyPathSepChar(sourcePath);
+                sourcePath.append(srcPath).trim();
+            }
+            const char * hostGroup = dropZone->queryProp("@hostGroup");
+            const char * host = "127.0.0.1";
+            if (hostGroup)
+            {
+               VStringBuffer xpath("storage/hostGroups[@name='%s']", srcPlane);
+               IPropertyTree * match = getGlobalConfigSP()->queryPropTree(xpath);
+               if (match)
+                   host = match->queryProp("hosts[1]");
```

Comment:
Needs some input from existing users to know if they actually use this feature in bare metal.

Reply:
For how since it is possible to define a plane without hosts, and/or there may be greater than one host,
perhaps it should error if there isn't exactly one hosts entry, e.g. if (1 != match->getCount("hosts")) ...

Reply:
@ghalliday mentioned that multiple host ip's are being used.  To support this,  both srcplane and srcip may be provided to dfuplus so that the speciifc host ip can be selected (from this list of ip's).  @jakesmith please see last commit.